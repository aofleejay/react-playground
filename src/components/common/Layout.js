import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

const Layout = (props) => (
  <Fragment>
    <header>Header</header>
    <div>{props.children}</div>
    <footer>Footer</footer>
  </Fragment>
)

Layout.propTypes = {
  children: PropTypes.node,
}

Layout.defaultProps = {
  children: null,
}

export default Layout
