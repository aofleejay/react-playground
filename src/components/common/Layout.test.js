import React from 'react'
import { shallow } from 'enzyme'
import Layout from './Layout'

describe('Test layout ui structure', () => {
  describe('Match its snapshot', () => {
    it('When all props exists', () => {
      const Child = () => <p>child component.</p>

      const wrapper = shallow(<Layout><Child /></Layout>)

      expect(wrapper).toMatchSnapshot()
    })
  
    it('When no props exists', () => {
      const wrapper = shallow(<Layout />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
