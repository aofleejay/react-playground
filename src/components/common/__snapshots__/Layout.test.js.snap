// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test layout ui structure Match its snapshot When all props exists 1`] = `
<React.Fragment>
  <header>
    Header
  </header>
  <div>
    <Child />
  </div>
  <footer>
    Footer
  </footer>
</React.Fragment>
`;

exports[`Test layout ui structure Match its snapshot When no props exists 1`] = `
<React.Fragment>
  <header>
    Header
  </header>
  <div />
  <footer>
    Footer
  </footer>
</React.Fragment>
`;
